pkg/core/native/crypto_blspoints.go

```diff
@@ -0,0 +1,84 @@
+package native
+
+import (
+	"fmt"
+
+	bls12381 "github.com/consensys/gnark-crypto/ecc/bls12-381"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+)
+
+// Wrappers around bls12381 types that must be used as stackitem.Interop values and
+// implement stackitem.Equatable interface.
+type (
+	G1Affine struct {
+		*bls12381.G1Affine
+	}
+	G1Jac struct {
+		*bls12381.G1Jac
+	}
+	G2Affine struct {
+		*bls12381.G2Affine
+	}
+	G2Jac struct {
+		*bls12381.G2Jac
+	}
+	GT struct {
+		*bls12381.GT
+	}
+)
+
+var (
+	_ = stackitem.Equatable(&G1Affine{})
+	_ = stackitem.Equatable(&G1Jac{})
+	_ = stackitem.Equatable(&G2Affine{})
+	_ = stackitem.Equatable(&G2Jac{})
+	_ = stackitem.Equatable(&GT{})
+)
+
+// Equals implements stackitem.Equatable interface.
+func (a *G1Affine) Equals(other any) bool {
+	b, ok := other.(*G1Affine)
+	return ok && a.Equal(b.G1Affine)
+}
+
+// Equals implements stackitem.Equatable interface.
+func (a *G1Jac) Equals(other any) bool {
+	b, ok := other.(*G1Jac)
+	return ok && a.Equal(b.G1Jac)
+}
+
+// Equals implements stackitem.Equatable interface.
+func (a *G2Affine) Equals(other any) bool {
+	b, ok := other.(*G2Affine)
+	return ok && a.Equal(b.G2Affine)
+}
+
+// Equals implements stackitem.Equatable interface.
+func (a *G2Jac) Equals(other any) bool {
+	b, ok := other.(*G2Jac)
+	return ok && a.Equal(b.G2Jac)
+}
+
+// Equals implements stackitem.Equatable interface.
+func (a *GT) Equals(other any) bool {
+	b, ok := other.(*GT)
+	return ok && a.Equal(b.GT)
+}
+
+// NewEquatable converts provided bls12381 point type to the stackitem.Equatable.
+func NewEquatable(point any) stackitem.Equatable {
```

Comment:
This may also help in avoiding interface with `any` here, you do type-switches in contract methods anyway, so a simple (unexported!) `type blsPoint` can be more convenient.

Reply:
Fixed, but currently it can't be unexported due to our tests that use neotest framework. They are located in a separate package and it's important for them to check the exact type of the point.

If we still would like to keep this structure unexported, then I can add `bls12381Serialize` call in the end of each test, but it will complicate them a bit. Implement?

Reply:
You can add `Bytes() []byte` into `BlsPoint` and cast `itm.(*stackitem.Interop).Value()` to an interface with `Bytes()`.

Reply:
Fixed.